src/formatDate.js

```diff
@@ -50,7 +50,44 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const modifyDate = date.split(fromFormat[fromFormat.length - 1]);
+  const newDate = [0, 0, 0];
+  const object = {};
+  let count = 0;
+
+  let year = 0;
+
+  for (const key of fromFormat) {
+    if (count <= modifyDate.length - 1) {
+      object[key] = modifyDate[count];
+    }
+    count += 1;
+  }
+
+  for (let i = 0; i < modifyDate.length; i++) {
+    for (const key in object) {
+      if (key === 'YYYY' && toFormat[i] === 'YY' && object[key].length === 4) {
```

Comment:
I may be wrong but it seem like the first condition (key === 'YYYY') is not needed if you als echeck object[key].length later

Reply:
Good spot! These checks actually check for the same thing